include/openthread/ip6.h

```diff
@@ -382,6 +382,23 @@ otError otIp6CreateSemanticallyOpaqueIid(otInstance *aInstance, otNetifAddress *
  */
 otMessage *otIp6NewMessage(otInstance *aInstance, const otMessageSettings *aSettings);
 
+/**
+ * Allocate a new message buffer and write the IPv6 datagram to the message buffer for sending an IPv6 message.
+ *
+ * @param[in]  aInstance             A pointer to an OpenThread instance.
+ * @param[in]  aData                 A pointer to the IPv6 datagram buffer.
+ * @param[in]  aDataLength           The size of the IPv6 datagram buffer pointed by @p aData.
+ * @param[in]  aLinkSecurityEnabled  TRUE if the message should be secured at Layer 2, FALSE otherwise.
+ *
+ * @returns A pointer to the message or NULL if malformed IPv6 header or insufficient message buffers are available.
+ *
+ * @sa otFreeMessage
+ */
+otMessage *otIp6NewMessageFromBuffer(otInstance *   aInstance,
+                                     const uint8_t *aData,
+                                     uint16_t       aDataLength,
+                                     bool           aLinkSecurityEnabled);
```

Comment:
How about using `otMessageSettings` and document that `mPriority` is ignored and obtained from the IPv6 message itself?

Reply:
Changed.